moped-editor/src/views/projects/projectView/ProjectComponents/TheMap.js

```diff
@@ -131,53 +112,61 @@ export default function TheMap({
     // });
   };
 
-  const onClick = (e) => {
-    if (e.features.length === 0) {
-      // clear clickedComponent if map is clicked elsewhere
-      if (clickedComponent) {
-        setClickedComponent(null);
-      }
-      return;
-    }
-
-    /* If not editing, handle click on project feature */
-    if (!isEditingComponent) {
-      const clickedProjectFeature = e.features.find((feature) =>
-        feature.layer.id.includes("project")
+  const handleCreateOnClick = (e) => {
```

Comment:
This is called from the `onClick` handler passed to the map (just like `handleEditOnClick` below). I think it is easier to separate out what is happening during component creation and component editing this way because it became a large chunk of code. It would be nice to get them out of here, but they rely on a lot of state that is in scope. 100% open to feedback on this